src/client/components/RegistrationConsents.tsx

```diff
@@ -23,48 +26,54 @@ const consentToggleCss = (noMarginBottom = false) => css`
 	gap: ${space[3]}px;
 `;
 
+const chooseNewsletter = (
+	geolocation: GeoLocation | undefined,
+	appName: AppName | undefined,
+): RegistrationNewsletterFormFields | undefined => {
+	const isFeast = appName === 'Feast';
+
+	if (isFeast) {
+		return RegistrationNewslettersFormFieldsMap.feast;
+	}
+
+	switch (geolocation) {
+		case 'GB':
+		case 'EU':
+		case 'ROW':
+			return RegistrationNewslettersFormFieldsMap.saturdayEdition;
+		case 'US':
+			return RegistrationNewslettersFormFieldsMap.usBundle;
+		case 'AU':
+			return RegistrationNewslettersFormFieldsMap.auBundle;
+		default:
+			// We want to show Saturday Edition even for an undefined location
+			return RegistrationNewslettersFormFieldsMap.saturdayEdition;
+	}
+};
+
 export const RegistrationConsents = ({
 	geolocation,
 	useIdapi,
 	noMarginBottom,
 	appName,
 }: RegistrationConsentsProps) => {
-	// check if the app is Feast or not
-	const isFeast = appName === 'Feast';
-
-	// don't show the Saturday Edition newsletter option for US and AUS or if using Feast app
-	const showSaturdayEdition = (() => {
-		const isValidLocation = !(['US', 'AU'] satisfies GeoLocation[]).some(
-			(location: GeoLocation) => location === geolocation,
-		);
-
-		return isValidLocation && !isFeast;
-	})();
-
-	// show the Feast newsletter option if the app is Feast and Saturday Edition is not shown
-	const showFeast = isFeast && !showSaturdayEdition;
-
+	const registrationNewsletter = chooseNewsletter(geolocation, appName);
 	// Show marketing consent if not showing Feast
-	const showMarketingConsent = !showFeast;
+	const showMarketingConsent =
+		!registrationNewsletter ||
+		registrationNewsletter !== RegistrationNewslettersFormFieldsMap.feast;
```

Comment:
i find this confusing to read, could we make a function similar to the `chooseNewsletter` above that determines if this should show, maybe even an iife?

```suggestion
	const showMarketingConsent =
		!registrationNewsletter ||
		registrationNewsletter !== RegistrationNewslettersFormFieldsMap.feast;
	const showMarketingConsent = (() => {
		if (registrationNewsletter === RegistrationNewslettersFormFieldsMap.feast) {
			return false;
		}
		return true;
	})();
```